pkg/api/customization/alert/notifier_action.go

```diff
@@ -42,15 +44,23 @@ func (h *Handler) testNotifier(actionName string, action *types.Action, apiConte
 		Message string
 		v3.NotifierSpec
 	}{}
+	clientNotifier := &struct {
+		client.NotifierSpec
+	}{}
+
 	if err = json.Unmarshal(data, input); err != nil {
-		return errors.Wrap(err, "unmarshaling input error")
+		return errors.Wrap(err, "unmarshalling input error")
+	}
+	if err = json.Unmarshal(data, clientNotifier); err != nil {
+		return errors.Wrap(err, "unmarshalling input error client")
 	}
+
 	notifier := &v3.Notifier{
 		Spec: input.NotifierSpec,
 	}
 	msg := input.Message
+	ns, id := ref.Parse(apiContext.ID)
 	if apiContext.ID != "" {
-		ns, id := ref.Parse(apiContext.ID)
```

Comment:
These variables are only used inside the condition block, is it a better style to move this out?